include/macros.h

```diff
@@ -18,4 +18,8 @@
 
 #define ABS(x) (((x) >= 0) ? (x): -(x))
 
+#define CLAMP(x, min, max) ((x) < (min) ? (min) : (x) > (max) ? (max) : (x))
+#define CLAMP_MAX(x, max) ((x) > (max) ? (max) : (x))
+#define CLAMP_MIN(x, min) ((x) < (min) ? (min) : (x))
```

Comment:
Why? They don't seem to be used

Reply:
I added them for testing the kq mis-match, but doesn't seemto hurt to add them anyway?

Reply:
The repo already has `MAX` and `MIN` macros which does the same as `CLAMP_MIN` and `CLAMP_MAX` respectively. There has been complains about the inverted names being confusing and there isn't really a need for them.
I guess the `CLAMP` macro can stay